packages/web/src/components/comments/CommentText.tsx

```diff
@@ -1,44 +1,141 @@
+import { ModalSource } from '@audius/common/models'
+import {
+  playerActions,
+  PurchaseableContentType,
+  trackPageLineupActions,
+  trackPageSelectors,
+  usePremiumContentPurchaseModal
+} from '@audius/common/store'
+import {
+  getDurationFromTimestampMatch,
+  timestampRegex
+} from '@audius/common/utils'
 import { Flex, Text, TextLink } from '@audius/harmony'
-import { useEffectOnce } from 'react-use'
+import { useDispatch, useSelector } from 'react-redux'
 
-const messages = {
-  seeMore: 'See More',
-  seeLess: 'See Less'
+import { UserGeneratedTextV2 } from 'components/user-generated-text/UserGeneratedTextV2'
+
+const { tracksActions } = trackPageLineupActions
+const { getLineup } = trackPageSelectors
+const { seek } = playerActions
+
+type TimestampLinkProps = {
+  timestamp: string
+  timestampSeconds: number
+}
+
+const TimestampLink = (props: TimestampLinkProps) => {
+  const { timestamp, timestampSeconds } = props
+
+  const dispatch = useDispatch()
+  const { track } = useCurrentCommentSection()
+  const lineup = useSelector(getLineup)
+  const { track_id: trackId } = track
+
+  const { hasStreamAccess } = useGatedContentAccess(track)
+
+  const uid = lineup?.entries?.[0]?.uid
+  const { onOpen: openPremiumContentPurchaseModal } =
+    usePremiumContentPurchaseModal()
+
+  return (
+    <TextLink
+      onClick={() => {
+        if (!hasStreamAccess) {
+          openPremiumContentPurchaseModal(
+            { contentId: trackId, contentType: PurchaseableContentType.TRACK },
+            {
+              source: ModalSource.Comment
+            }
+          )
+        } else {
+          dispatch(tracksActions.play(uid))
+          setTimeout(() => {
+            dispatch(seek({ seconds: timestampSeconds }))
+          })
+        }
+      }}
+      variant='visible'
+      size='s'
+    >
+      {timestamp}
+    </TextLink>
+  )
 }
 
 export type CommentTextProps = {
   children: string
   isEdited?: boolean
 }
 
-export const CommentText = ({ children }: CommentTextProps) => {
+export const CommentText = (props: CommentTextProps) => {
+  const { children, isEdited } = props
   const textRef = useRef<HTMLElement>()
   const [isOverflowing, setIsOverflowing] = useState(false)
   const [isExpanded, setIsExpanded] = useState(false)
+  const {
+    track: { duration }
+  } = useCurrentCommentSection()
 
-  useEffectOnce(() => {
+  useEffect(() => {
     setIsOverflowing(
       (textRef.current &&
         textRef.current.offsetHeight < textRef.current.scrollHeight) ||
         false
     )
-  })
+  }, [children])
 
   return (
     <Flex direction='column' alignItems='flex-start' gap='xs'>
-      <Text
-        // Issue with the HTMLElement ref here and the ref type for Text component
-        // @ts-ignore
-        ref={textRef}
-        variant='body'
-        size='s'
-        color='default'
-        maxLines={isExpanded ? undefined : 3}
-      >
-        {children}
-      </Text>
+      <p css={{ textAlign: 'left' }}>
```

Comment:
why the p tag here?

Reply:
Oh yeah I should probably just do as='p' on the UserGeneratedTextV2, good call